strategy/live_pairs.py

```
@@ -0,0 +1,452 @@
+    """Read held size per token from the Data API.
+
+    An independent view of what the venue says we hold. The registry records
+    what we believe; reconciling the two catches a class of bug that neither
+    source can catch alone.
+
+    Raises on any failure. An unreadable positions endpoint is not an empty
+    portfolio, and treating it as one would let the divergence check pass by
+    knowing nothing.
+    """
+    url = f"{DATA_API_BASE}/positions?user={funder}"
+    req = urllib.request.Request(url, headers={"User-Agent": "spread-hunter"})
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)